src/EntryPoints/SpecialNetwork.php

```diff
@@ -22,108 +27,169 @@ public function __construct() {
 	 * @param string|null $subPage
 	 */
 	public function execute( $subPage ) : void {
-		$request = $this->getRequest();
-		$output = $this->getOutput();
 		$this->setHeaders();
 
 		$config = new NetworkConfig();
-		$params = [];
-
-		$params[] =
-			'pages=' .
-			strtr( $request->getText( 'pages', Title::newMainPage()->getPrefixedText() ), "\n", '|' );
+		$params = $this->parseParams( $this->getRequest(), $config );
+		if ( $this->getRequest()->getCheck( 'pages' ) ) {
+			$this->getOutput()->addHTML( $this->showGraph( $this->formatParams( $params ), $config ) );
+		}
 
-		$exclude = $request->getText( 'exclude' );
-		if ( $exclude !== '' ) {
-			$params[] = 'exclude=' . strtr( $exclude, "\n", ';' );
+		if ( !$this->including() ) {
+			$this->showForm( $params );
 		}
+	}
 
-		$class = $request->getText( 'class' );
-		if ( $class !== '' ) {
-			$params[] = 'class=' . $class;
+	/**
+	 * @param WebRequest $request
+	 * @param NetworkConfig $config
+	 * @return string[] $params
```

Comment:
Not an issue - personally I'd omit the parameters since now the info is duplicated